### PR TITLE
make `trigger::TriggerEvent::filterLabel` return `std::string_view`

### PR DESCRIPTION
**Pull request by missirol**
#### PR description:

This PR tries to implement the suggestion in https://github.com/cms-sw/cmssw/issues/42672#issuecomment-1695959141 (and https://github.com/missirol/cmssw/commit/9550c30b26dca6d500a009bc6f08449132c5dd5a#r125745581), in order to avoid unnecessary memory allocations when trying to extract the label of the `edm::InputTag` associated to a certain `trigger::TriggerEvent::TriggerFilterObject` of an instance of `trigger::TriggerEvent`.

 - The return type of the method `trigger::TriggerEvent::filterLabel` is changed to `std::string_view`.
 - In connection to the specific issue mentioned in https://github.com/cms-sw/cmssw/issues/42672#issuecomment-1695959141, the plugin `AlCaHcalIsotrkProducer` is also modified accordingly, in order to reduce extra memory allocations.
 - A small, related, technical change to `PATTriggerProducer` is also included, see https://github.com/cms-sw/cmssw/pull/42675#issuecomment-1697071803.

#### PR validation:

None.

#### If this PR is a backport, please specify the original PR and why you need to backport that PR. If this PR will be backported, please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by missirol** [comment]
FYI @cms-sw/core-l2 , for your review.

**Comment by missirol** [comment]
please test

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalIsotrkProducer.cc [hunk 1]
```diff
@@ -654,7 +654,7 @@ void AlCaHcalIsotrkProducer::produce(edm::Event& iEvent, edm::EventSetup const&
             //loop over all trigger filters in event (i.e. filters passed)
             for (unsigned int ifilter = 0; ifilter < triggerEvent.sizeFilters(); ++ifilter) {
               std::vector<int> Keys;
-              std::string label = triggerEvent.filterTag(ifilter).label();
+              auto const label = triggerEvent.filterLabel(ifilter);
               //loop over keys to objects passing this filter
               for (unsigned int imodule = 0; imodule < moduleLabels.size(); imodule++) {
                 if (label.find(moduleLabels[imodule]) != std::string::npos) {
```
Comment (Martin-Grunewald) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalIsotrkProducer.cc:660]: Well, Chris's comment would also apply here as ```label`` is now a ```string_view``` - no longer a ```string``` with ```std::string::npos```... and anywhwere the ```findLabel``` is now used like this. ?

**Comment by missirol** [comment]
The latest commit (dc1a1a79aa5) just changes a call from `filterTag(idx).label()` to `filterLabel(idx)` in `PATTriggerProducer` (changing more of `PATTriggerProducer` to use `std::string_view` rather than `std::string` is beyond the scope of this PR).

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalIsotrkProducer.cc (hunk 1 above)
Comment (missirol) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalIsotrkProducer.cc]: Thanks, I fixed this in 6b7396d344b. I didn't find other instances where `filterLabel(idx)` is used which would require further changes.

**Comment by missirol** [comment]
+hlt

**Comment by missirol** [comment]
@perrotta , iirc, yesterday at the ORP it was mentioned that a backport of this PR might be useful. Is that the case ? (if so, to which cycles ?)

**Comment by perrotta** [comment]
> @perrotta , iirc, yesterday at the ORP it was mentioned that a backport of this PR might be useful. Is that the case ? (if so, to which cycles ?)

@missirol one should evaluate the amount of memory saved thanks to this PR. Then, based on it, one can judge whether it is worth to backport in releases still in production

**Comment by perrotta** [comment]
+alca

**Comment by missirol** [comment]
> one should evaluate the amount of memory saved thanks to this PR.

@makortel , could you please give your opinion ? I haven't understood the quantitative impact of this PR.

**Comment by makortel** [comment]
> > one should evaluate the amount of memory saved thanks to this PR.
> 
> could you please give your opinion ? I haven't understood the quantitative impact of this PR.

This PR is about reducing memory churn. There may be some impact in RSS, but I'd expect the main effect to be in reduced CPU usage. But it is difficult to say how much without explicitly testing. I'd _guess_ the impact would be visible (from the profiles shown in https://github.com/cms-sw/cmssw/issues/42672 this case alone caused about 7 % of the memory allocations during the event processing), but not a game changer. I also don't know the exact circumstances where this AlCaRECO(?) module is included in the data reprocessing campaigns.

**Comment by missirol** [comment]
Thanks for the clarification. If somebody thinks that backporting this is useful, I'm happy to do it. Otherwise, I will just leave this for `master`.

**Comment by missirol** [comment]
@cms-sw/reconstruction-l2 @cms-sw/xpog-l2 , kind ping to review this simple PR.

**Comment by simonepigazzini** [comment]
+1

**Comment by mandrenguyen** [comment]
+1

**Comment by rappoccio** [comment]
+1

**Pull request merged**